### PR TITLE
Cdo: init at 1.7.2

### PR DESCRIPTION
**Pull request by ltavard**
###### Motivation for this change
CDO is a collection of command line Operators to manipulate and analyse Climate and NWP model Data. 
We want to deploy it on our HPC platform.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/cdo/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{ stdenv, fetchurl, curl
+, enable_data ? true        # DATA support [default=yes]
```
Comment (joachifm) [inline pkgs/development/libraries/cdo/default.nix]: Note that all these parameters are not discoverable. Packages in Nixpkgs are usually configured with lowest-common denominator in mind, using params only sparingly. Reifying every `configure` switch as a parameter is certainly going overboard in any case.

Please consider reducing this to only cover options users are likely to want to change or for features that are particularly costly in terms of build footprint.

File: pkgs/development/libraries/cdo/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ++ stdenv.lib.optional with_hdf5 [ hdf5 ];
+
+  src = fetchurl {
+    url = "https://code.zmaw.de/attachments/download/12760/${name}.tar.gz";
+    sha256 = "4c43eba7a95f77457bfe0d30fb82382b3b5f2b0cf90aca6f0f0a008f6cc7e697";
+  };
+
+  # Configure phase 
+  configureFlags = ''
+    ${if enable_data then       "--enable-data" else ""}
```
Comment (joachifm) [inline pkgs/development/libraries/cdo/default.nix]: Please consider using `optionalString` or specifying `configureFlags` as a list of strings, using `optional(s)` for optional flags.

File: pkgs/development/libraries/cdo/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ${if enable_cdi_lib then    "--enable-cdi-lib" else ""}
+    ${if enable_all_static then "--enable-all-static" else ""}
+    ${if enable_cxx then        "--enable-cxx" else ""}
+    ${if with_gnu_ld then       "--with-gnu-ld" else ""}
+    ${if with_hdf5 then         "--with-hdf5=${hdf5}" else "--with-hdf5=no"}
+    ${if with_netcdf then       "--with-netcdf=${netcdf}" else "--with-netcdf=no"}
+    '';
+
+  meta = {
+    description = "collection of command line Operators to manipulate and analyse Climate and NWP model Data";
```
Comment (joachifm) [inline pkgs/development/libraries/cdo/default.nix]: Please capitalize `meta.description`.

File: pkgs/development/libraries/cdo/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    longDescription = '' Supported data formats are GRIB 1/2, netCDF 3/4, SERVICE, EXTRA and IEG.
```
Comment (joachifm) [inline pkgs/development/libraries/cdo/default.nix]: A better more idiomatic syntax for `longDescription` (or multi-line Nix strings in general) is
```
''
  Lorem ipsum
  Lorem ipsum.
''
```

**Comment by joachifm** [comment]
Also please consider building this with sandboxing enabled to help ensure that there are no implicit dependencies.

**Comment by ltavard** [comment]
I adjusted my default.nix file according to your  remarks.
I hope that answered your expectations.

File: pkgs/development/libraries/cdo/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchurl, curl, hdf5, netcdf
+, enable_cdi_lib ? false    # build, install and link to a CDI library [default=no]
+, enable_all_static ? false # build a completely statically linked CDO binary  [default=no]
+, enable_cxx ? false        # Use CXX as default compiler [default=no]
+, with_gnu_ld ? false       # assume the C compiler uses GNU ld [default=no]
```
Comment (joachifm) [inline pkgs/development/libraries/cdo/default.nix]: Is it actually necessary to specify `--with-gnu-ld`? I can't recall that ever being an issue, autoconf seems to be able to figure it out.

Also see other expressions that use feature flags; you normally use parameters named `fooSupport` or `withFoo`. For consistency, I recommend following that pattern.

Comment (ltavard) [inline pkgs/development/libraries/cdo/default.nix]: Thanks for the comment, after testing it's not necessary. So I deleted the option

File: pkgs/development/libraries/cdo/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    url = "https://code.zmaw.de/attachments/download/12760/${name}.tar.gz";
+    sha256 = "4c43eba7a95f77457bfe0d30fb82382b3b5f2b0cf90aca6f0f0a008f6cc7e697";
+  };
+
+ # Configure phase 
+ configureFlags = [
+   "--with-netcdf=${netcdf}" "--with-hdf5=${hdf5}"]
+   ++ stdenv.lib.optional (enable_cdi_lib) "--enable-cdi-lib"
+   ++ stdenv.lib.optional (enable_all_static) "--enable-all-static"
+   ++ stdenv.lib.optional (enable_cxx) "--enable-cxx"
```
Comment (joachifm) [inline pkgs/development/libraries/cdo/default.nix]: Syntax error here, missing list terminator.

Comment (ltavard) [inline pkgs/development/libraries/cdo/default.nix]: Sorry for this,
I have just pushed the new version

**Comment by knedlsepp** [comment]
@ltavard: Could you squash your changes to a single `cdo: init at 1.7.2` commit? I think this is pretty clean now and we should get this merged. (I wanna use it! ;-) )
@joachifm: Is there anything more that you think should be done before we can merge it?
One more thing: Just did a quick compilation test and it should work on macOS too:
```
platforms = with stdenv.lib.platforms; linux ++ darwin;
```

**Comment by zimbatm** [comment]
All good! Just added darwin and fixed some ws and then merged

**Pull request merged**